src/gj.rs

```diff
@@ -247,7 +249,7 @@ impl<'b> Context<'b> {
                             let i = self.query.vars.get_index_of(v).unwrap();
                             self.tuple[i]
                         }
-                        AtomTerm::Value(val) => *val,
+                        AtomTerm::Literal(lit) => self.egraph.eval_lit(lit),
```

Comment:
isn't evaluating literals in the query a little slower and indirect than storing the value itself?